### ISSUE TITLE
Room - App opens random chat after leaving public room when logging in via public room link

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#42522]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.75-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Found when validating PR :** https://github.com/Expensify/App/pull/42020
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to staging.new.expensify.com
2. Log out if logged in.
3. Navigate to public room link.
4. Click Sign in on the bottom banner.
5. Log in with an existing account.
6. After login, without opening other reports, leave the public room.

## Expected Result:
App will redirect to Concierge chat after leaving public room when user signs in via public room link.

## Actual Result:
App redirects to a random chat after leaving public room when user signs in via public room link.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/b8bce46f-41ae-49e2-9a4b-2f859ff4a2fb

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
@puneetlath I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
We think this issue might be related to the #vip-vsb.

**Comment by puneetlath** [comment]
I don't think we need to block the deploy over this. It's a very specific scenario.

**Comment by ahmedGaber93** [comment]
@puneetlath I think this expected, the user is redirect to last opened report not to random report and this was the expected result when we were working on https://github.com/Expensify/App/issues/41128

https://github.com/Expensify/App/issues/41128 expected result:
1. if users leave a room or group, we will redirect them to the last opened report (sort onyx reports by last read and find the last one)
2. if a new user open the room from deep link, so he didn't open any report yet, we will redirect them to concierge

CC @Julesssss @Nodebrute
